checker/src/main/java/org/checkerframework/checker/nullness/messages.properties

```
@@ -22,6 +22,7 @@ nulltest.redundant=redundant test against null; "%s" is non-null
 instanceof.nullable=instanceof is only true for a non-null expression
 instanceof.nonnull.redundant=redundant @NonNull annotation on instanceof
 new.array.type.invalid=annotations %s may not be applied as component type for array "%s"
+new.array.nullable.ignored=nullable annotation on new array main modifier ignored, must be nonnull
```

I don't see any other message, here or in the basetype, refer to the `main` modifier... which is odd... somewhere we called it `primary`, but I also don't see that in any of the error messages... Can you look what would be a consistent way to word this?

Other messages in this file also don't use `nullable` and use `non-null`.

---

I also can't see any reference to main or primary anywhere else. Where does this wording originate from, what other contexts is it used in?

We could use the [JLS](https://docs.oracle.com/javase/specs/jls/se23/html/jls-9.html#jls-9.7.4) wording:

```suggestion
new.array.nullable.ignored=array type annotation must be nonnull
```